torch/masked/maskedtensor/creation.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+
+import torch
+
+from .core import MaskedTensor
+
+
+# Basic factory function
+def masked_tensor(data, mask, requires_grad=False):
+    from .core import is_masked_tensor
+
+    assert not is_masked_tensor(data)
+    assert not is_masked_tensor(mask)
+    data = data.clone().detach()
```

Comment:
Why clone?

Reply:
I think the original thinking was that we wanted new memory for the data as opposed to just a different instance? Do you think that's fair or is the clone unnecessary?